clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -887,6 +887,11 @@ llvm::DIType *CGDebugInfo::CreateType(const BuiltinType *BT) {
   case BuiltinType::SatULongFract:
     Encoding = llvm::dwarf::DW_ATE_unsigned_fixed;
     break;
+  case BuiltinType::DecimalFloat32:
+  case BuiltinType::DecimalFloat64:
+  case BuiltinType::DecimalFloat128:
+    Encoding = llvm::dwarf::DW_ATE_unsigned;
```

Comment:
```suggestion
    Encoding = llvm::dwarf::DW_ATE_decimal_float;
```
The DWARF specification already defines an encoding value for the IEEE 754R DFP types and an identifier is present in `llvm/include/llvm/BinaryFormat/Dwarf.def`, so we can set the right encoding now. I'm not sure if differentiation is required for BID vs DPD encoding here, but we can revisit that as part of https://github.com/tahonermann/llvm-project/issues/20 later.